### DISCUSSION TITLE
When created using three levels of nesting, the innermost level with the outermost foreign key will fail to create

### DISCUSSION THREAD

**Discussion by myNameIsDu** [prisma/prisma#14423] [Q&A]
Here is an example
```prisma
model TestPage {
  id                        Int                         @id @default(autoincrement())
  pageName                  String                      @db.VarChar(20)
  TestPageComponent         TestPageComponent[]
  TestPageComponentQuestion TestPageComponentQuestion[]
}

model TestPageComponent {
  id                        Int                         @id @default(autoincrement())
  pageId                    Int
  page                      TestPage                    @relation(fields: [pageId], references: [id])
  componentName             String                      @db.VarChar(20)
  TestPageComponentQuestion TestPageComponentQuestion[]
}

model TestPageComponentQuestion {
  id           Int               @id @default(autoincrement())
  componentId  Int
  component    TestPageComponent @relation(fields: [componentId], references: [id])
  pageId       Int
  page         TestPage          @relation(fields: [pageId], references: [id])
  questionName String            @db.VarChar(20)
}
```

```typescript
    await this.prisma.testPage.create({
      data: {
        pageName: 'xxx',
        TestPageComponent: {
          create: [
            {
              componentName: 'component',
              TestPageComponentQuestion: {
                create: [
                  {
                    questionName: '123',
                  },
                ],
              },
            },
          ],
        },
      },
    });
```

And then it gave me an error
```shell
Argument page for data.TestPageComponent.create.0.TestPageComponentQuestion.create.0.page is missing.
```

**Comment by pantharshit00** [comment]
Hello @myNameIsDu , our apologies for the delay in getting back to you! Due to the large number of questions in our community channels, unfortunately this slipped through the cracks. It's very important for us to address your question, so let us know in a comment if this is still an open discussion. 

Looks like you are not providing all the required fields in the second create field. You will need to do the following:
```ts
const data = await prisma.testPage.create({
    data: {
      pageName: "xxx",
      TestPageComponent: {
        create: [
          {
            componentName: "component",
            TestPageComponentQuestion: {
              create: [
                {
                  questionName: "123",
                  page: "page",
                  pageId: "12",
                },
              ],
            },
          },
        ],
      },
    },
  });
```

**Reply by myNameIsDu** [reply]
Thanks for your reply, But the key to the question is when using this `create` sentence, I don't know the `pageId` and `page`, I think it can be smarter, When using this `create` sentence, `TestPageComponentQuestion` can be created automatically associated with the current page. In the end, You do a great job and thank you for all you do,

**Reply by pantharshit00** [reply]
With "currentPage", which model you are referring to?

**Reply by myNameIsDu** [reply]
The "currentPage" I meant that's creating the page, Since that's creating, so i don't know its `page` and `pageId`. so I want to it can be smarter.

**Reply by pantharshit00** [reply]
Sorry, I wanted to ask which model you are referring to which are either of `TestPage`, `TestPageComponent` and `TestPageComponentQuestion`. They are similarly named so it is quite hard to follow things here.

**Reply by pantharshit00** [reply]
Hello @myNameIsDu,

Just following up to know if this is still relevant. You didn't answer my query last time so I wasn't able to proceed here.

**Reply by myNameIsDu** [reply]
I'm so sorry, I missed the email. Thank you for replying to me. And I think I have to explain my question again, sorry, My native language isn't English, So maybe it's not very exact.

When I used the model below

```prisma
model TestPage {
  id                        Int                         @id @default(autoincrement())
  pageName                  String                      @db.VarChar(20)
  TestPageComponent         TestPageComponent[]
  TestPageComponentQuestion TestPageComponentQuestion[]
}

model TestPageComponent {
  id                        Int                         @id @default(autoincrement())
  pageId                    Int
  page                      TestPage                    @relation(fields: [pageId], references: [id])
  componentName             String                      @db.VarChar(20)
  TestPageComponentQuestion TestPageComponentQuestion[]
}

model TestPageComponentQuestion {
  id           Int               @id @default(autoincrement())
  componentId  Int
  component    TestPageComponent @relation(fields: [componentId], references: [id])
  pageId       Int
  page         TestPage          @relation(fields: [pageId], references: [id])
  questionName String            @db.VarChar(20)
}
```
The relation is `TestPage => TestPageComponent => TestPageComponentQuestion`, But there's another relation here. `TestPage => TestPageComponentQuestion`.  So when I use the `create` sentence.

```prisma
    await this.prisma.testPage.create({
      data: {
        pageName: 'xxx',
        TestPageComponent: {
          create: [
            {
              componentName: 'component',
              TestPageComponentQuestion: {
                create: [
                  {
                    questionName: '123',
                  },
                ],
              },
            },
          ],
        },
      },
    });
```

And then it told me an error:

```prisma
Argument page for data.TestPageComponent.create.0.TestPageComponentQuestion.create.0.page is missing.
```
But the key is that the page is the one I am currently creating. I don't its `pageId`. So I think it can be smarter.

And I think this problem is very common. If I only have the relation `TestPage => TestPageComponent => TestPageComponentQuestion`. then I want to know how many `questions` are on which `page`. I have got to search how many `components` are on which `page` first. if I have the relation `TestPage => TestPageComponentQuestion`. I can directly search how many`questions` are on which `page`. It's useful.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
